jmix-security/security-ui/src/main/java/io/jmix/securityui/authentication/LoginScreenSupport.java

```diff
@@ -169,11 +169,30 @@ public Authentication authenticate(AuthDetails authDetails,
                         authDetails.getTimeZone())
         );
 
+        preventSessionFixation(authenticationToken);
+
         onSuccessfulAuthentication(authenticationToken, authDetails, frameOwner);
 
         return authenticationToken;
     }
 
+    protected void preventSessionFixation(Authentication authentication) {
+        if (authentication.isAuthenticated()
+                && VaadinRequest.getCurrent() != null
+                && uiProperties.isUseSessionFixationProtection()) {
+            VaadinService.reinitializeSession(VaadinRequest.getCurrent());
+
+            WrappedSession session = VaadinSession.getCurrent().getSession();
+            int timeout = uiProperties.getHttpSessionExpirationTimeoutSec();
+            session.setMaxInactiveInterval(timeout);
+
+            HttpSession httpSession = session instanceof WrappedHttpSession ?
+                    ((WrappedHttpSession) session).getHttpSession() : null;
+            log.info("Session reinitialized: HttpSession={}, timeout={}sec",
```

Comment:
First, do we need this log message with "info" level? Right now we don't log the fact the user has been logged in, but log some "Session reinitialzed" message.
Another thing - if the session is not WrappedHttpSession we will log "Session reinitialized: HttpSession=null" - is this a desired behavior?

Reply:
Regarding `HttpSession`, it appears that `WrappedSession`  has 2 implementations, both have different methods to obtain original session, and in case of `LoginScreenSupport` its unlikely that we get `WrappedPortletSession`, but `WrappedSession` has no redeable `toString()` implementation.

Reply:
As I see, `WrappedSession` has the `getId()` method. Isn't it what we want to see in logs?